components/public-api/go/protoc-proxy-gen/protoc-proxy-gen.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2023 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package main
+
+import (
+	"fmt"
+	"path"
+
+	"google.golang.org/protobuf/compiler/protogen"
+	"google.golang.org/protobuf/types/pluginpb"
+)
+
+const (
+	contextPackage = protogen.GoImportPath("context")
+	connectPackage = protogen.GoImportPath("github.com/bufbuild/connect-go")
+)
+
+func main() {
+	protogen.Options{}.Run(func(gen *protogen.Plugin) error {
+		gen.SupportedFeatures = uint64(pluginpb.CodeGeneratorResponse_FEATURE_PROTO3_OPTIONAL)
+
+		for _, f := range gen.Files {
+			if !f.Generate {
+				continue
+			}
+			generateFile(gen, f)
+		}
+		return nil
+	})
+}
+
+func generateFile(gen *protogen.Plugin, file *protogen.File) {
+	// We only generate our proxy implementation for services, not for raw structs
+	if len(file.Services) == 0 {
+		return
+	}
+
+	var (
+		targetPackageName = fmt.Sprintf("%sconnect", file.GoPackageName)
+
+		filename = path.Join(
+			path.Dir(file.GeneratedFilenamePrefix),
+			targetPackageName,
+			fmt.Sprintf("%s.proxy.connect.go", path.Base(file.GeneratedFilenamePrefix)))
+		importPath = protogen.GoImportPath(path.Join(string(file.GoImportPath), string(file.GoPackageName)))
+	)
+
+	// Setup a new generated file
+	g := gen.NewGeneratedFile(filename, importPath)
+
+	// generate preamble
+	g.P("// Code generated by protoc-proxy-gen. DO NOT EDIT.")
+	g.P()
+	g.P("package ", targetPackageName)
+	g.P()
+	g.Import(file.GoImportPath)
+	g.P()
+
+	// generate individual services
+	for _, service := range file.Services {
+		// generate struct definition
+		handlerStructName := "Proxy" + service.GoName + "Handler"
+
+		// Generate a type assertion to ensure the handler implements the connect handler interface
+		g.P("var _ " + service.GoName + "Handler" + " = (*" + handlerStructName + ")(nil)")
```

Comment:
nit: for consistencies sake fmt.Sprintf would be preferable throughout

Reply:
done